modules/ml/include/opencv2/ml.hpp

```diff
@@ -1912,75 +1912,47 @@ class CV_EXPORTS_W ANN_MLP_ANNEAL : public ANN_MLP
 *                                   Simulated annealing solver                             *
 \****************************************************************************************/
 
-/** @brief The class defines interface for system state used in simulated annealing optimization algorithm.
-
-@cite Kirkpatrick83 for details
+#ifdef CV_DOXYGEN
+/** @brief This class declares example interface for system state used in simulated annealing optimization algorithm.
 */
-class CV_EXPORTS SimulatedAnnealingSolverSystem
+struct SimulatedAnnealingSolverSystem
 {
-protected:
-    inline SimulatedAnnealingSolverSystem() {}
-public:
-    virtual ~SimulatedAnnealingSolverSystem() {}
-
     /** Give energy value for a state of system.*/
-    virtual double energy() const = 0;
+    double energy() const;
     /** Function which change the state of system (random pertubation).*/
-    virtual void changeState() = 0;
+    void changeState();
     /** Function to reverse to the previous state. Can be called once only after changeState(). */
-    virtual void reverseState() = 0;
+    void reverseState();
 };
+#endif // CV_DOXYGEN
 
 /** @brief The class implements simulated annealing for optimization.
- *
-@cite Kirkpatrick83 for details
-*/
-class CV_EXPORTS SimulatedAnnealingSolver : public Algorithm
-{
-public:
-    SimulatedAnnealingSolver(const Ptr<SimulatedAnnealingSolverSystem>& system);
-    inline ~SimulatedAnnealingSolver() { release(); }
 
-    /** Simulated annealing procedure. */
-    int run();
-    /** Set/initialize RNG (energy).
-    @param rng new RNG
-    */
-    void setEnergyRNG(const RNG& rng);
-    /** Set initial temperature of simulated annealing procedure.
-    @param x new initial temperature. x\>0
-    */
-    void setInitialTemperature(double x);
-    /** Set final temperature of simulated annealing procedure.
-    @param x new final temperature value. 0\<x\<initial temperature
-    */
-    void setFinalTemperature(double x);
-    /** Get final temperature of simulated annealing procedure. */
-    double getFinalTemperature();
```

Comment:
Can we get final temperature after stopped algorithm? To restart algorithm final temperature is necessary

Reply:
yes, see "lastTemperature" optional CV_OUT parameter

BTW, There are no automatic bindings, because custom class SimulatedAnnealingSolverSystem is used.

Reply:
@LaurentBerger Do you have any objections about interface update? Does existed interface mentioned in your [book](https://opencv.org/books.html)?

Reply:
That's ok with CV_OUT. I don't use Simulated Annealing in my book so there is no problem.

There is only problem it is about regression in class ann_mlp. RPROP optimize networks : it is regression. SimulatedAnnealingANN_MLP uses calcError to optimize network and it is not regression, it is a classifier (except when there is only neuron in output layer) . It is not possible in interface to set classifier to true  (=regression).
When it is regression and there is more than 2 neurons in output layer SimulatadeAnnealingANN_MLP calcerror must give this results : 

float val = s.predict(sample, respRow);
err += norm(respRow - responses.row(si));